apps/cowswap-frontend/src/utils/getExecutedSummaryData.ts

```diff
@@ -19,135 +18,31 @@ export interface ExecutedSummaryData {
   swappedAmountWithFee: BigNumber
 }
 
-export function getExecutedSummaryData(order: GenericOrder): ExecutedSummaryData {
-  const { inputToken, outputToken } = order
-
-  const parsedInputToken = new Token(
-    inputToken.chainId,
-    inputToken.address,
-    inputToken.decimals,
-    inputToken.symbol,
-    inputToken.name,
-  )
-  const parsedOutputToken = new Token(
-    outputToken.chainId,
-    outputToken.address,
-    outputToken.decimals,
-    outputToken.symbol,
-    outputToken.name,
-  )
-
-  const surplusToken = isSellOrder(order.kind) ? parsedOutputToken : parsedInputToken
-  return getExecutedSummaryDataWithSurplusToken(order, surplusToken)
-}
+export function getExecutedSummaryData(order: GenericOrder, intermediateToken: Nullish<Token>): ExecutedSummaryData {
```

Comment:
A bit of context for #6629 fix.

1. Check `resolveDisplayTokens` function and you will see that `defaultSurplusToken` is redundant, because in case of buy order we exit from the function with:
```
{
  effectiveOutputToken: parsedOutputToken,
  surplusDisplayToken: parsedInputToken,
  surplusAmount: CurrencyAmount.fromRawAmount(parsedInputToken, rawSurplus),
}
```
2. In the sell order case, we can replace `defaultSurplusToken` with `parsedOutputToken`

<img width="1720" height="1816" alt="image" src="https://github.com/user-attachments/assets/09631bf2-c94b-418c-9b42-19710f123387" />